src/main/scala/ie/zalando/fabric/gateway/service/IngressDerivationChain.scala

```diff
@@ -213,14 +228,37 @@ class IngressDerivationChain(stackSetOpertions: StackSetOperations)(implicit mat
     }
   }
 
+  def genCorsPreflightRoute(route: Route,
+                            gatewayContext: GatewayContext,
+                            allowedOrigins: Set[Uri],
+                            allowedMethods: Set[HttpVerb]): SkipperRouteDefinition = {
+    val predicates = NEL.of(route.path, MethodMatch(route.verb), HttpsTraffic)
+    val filters = NEL.of(
+      EnableAccessLog(List(4, 5)),
+      Status(204),
+      FlowId,
+      CorsOrigin(allowedOrigins),
+      ResponseHeader("\"Access-Control-Allow-Methods\"", s""""${allowedMethods.map(_.value).mkString(", ")}""""),
+      ResponseHeader("\"Access-Control-Allow-Headers\"", "\"X-Flow-Id, Content-Type\""),
```

Comment:
Does this mean that all other headers will be blocked by the browser? Should we be doing this as * maybe?

Reply:
Or is that header definitely required?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers 

It's required if the request has `Access-Control-Request-Headers`, again like the other response headers we're in the situation where we can't optionally include headers so we're just throwing it in there. If we want to do this properly we need to add a more complete cors configuration filter to skipper.